### ISSUE TITLE
Add `-DefaultValue` to Get-Item and Get-Variable

### ISSUE THREAD

**Issue by dkaszews** [PowerShell/PowerShell#17960]
### Summary of the new feature / enhancement

PowerShell 7 introduces null-coallescing operator `??` and its compound assignement counterpart `??=`. They can be used to conveniently provide default value for variables which may be null. It is however also possible for a variable to not be set at all, in which case `??` and `??=` throw a `StrictMode` error if enabled. This is often the case when using environment variables.

The current way to write it is a quite verbose `(Test-Path 'variable:x') ? $x : $y`, which is similar pattern to `$x -eq $null ? $x : $x` which `??` replaces, and which is usable when `StrictMode` is not enabled. This may tempt the user to disable `StrictMode` to write more compact code.

Proposition is to add `-DefaultValue` to `Get-Item` and `Get-Variable`, which transforms the code above into much more pleasant and readable `Get-Item 'variable:x' -Default $y` or `Get-Variable 'x' -Default $y`, avoiding repetition of the variable name.

The null compound assignment `??=` could be replicated by adding `-SkipIfExists` to `Set-Variable`, but this is beyond scope of this issue.

**Comment by 237dmitry** [comment]
The idea is not quite clear in the context of this syntax. Are you suggesting something similar?:

```
(Get-item variable:x -ea 0) ? $x : ($x = 10)
```

**Comment by dkaszews** [comment]
> The idea is not quite clear in the context of this syntax. Are you suggesting something similar?:
> 
> ```
> (Get-item variable:x -ea 0) ? $x : ($x = 10)
> ```

I don't follow, how is this different from the `(Test-Path 'variable:x') ? $x : $y` example I provided? Other than using `-ErrorAction SilentlyContinue` disguised under `-ea 0` completely hides the intent, while my proposal is to clearly state intent of providing a default value.

**Comment by 237dmitry** [comment]
> while my proposal is to clearly state intent of providing a default value.

>> The idea is not quite clear in the context of this syntax.

I was referring specifically to this syntax:  `a ? b : c`
Could you give a conceptual example?

**Comment by dkaszews** [comment]
Maybe a more real life example would be clearer. In my script, I want to be able to override the default settings by setting an environment variable, as is common on Linux. Use of what is essentially a global may not be great, but in case of scripts it is usually the easiest way to go.

Currently I have written the following:

```pwsh
$RetryCount = (Test-Path 'env:RETRY_COUNT') ? $env:RETRY_COUNT : 3
$RetryTimeout = (Test-Path 'env:RETRY_TIMEOUT') ? $env:RETRY_TIMEOUT : 5000
./run_tests.py -- repeat $RetryCount --timeout $RetryTimeout
```

It's quite readable, as long as you know that `Test-Path` can be used for variables, but requires typing the environment variable name twice. With my proposal, this can nicely fit on a single line while being even more clear:

```pwsh
./run_tests.py -- repeat (Get-Item 'env:RETRY_COUNT' -Default 3) --timeout (Get-Item 'env:RETRY_TIMEOUT' -Default 5000)
```

**Comment by 237dmitry** [comment]
> Maybe a more real life example would be clearer

Yes, this explanation is quite clean. Perhaps it would be useful, but I am a supporter of the usual syntax with conditions and their checks even if it is longer.

**Comment by dkaszews** [comment]
Still an issue, shut up bot
